src/main/java/org/alliancegenome/curation_api/services/associations/alleleAssociations/AlleleGeneAssociationService.java

```diff
@@ -80,20 +91,40 @@ public List<Long> getAlleleGeneAssociationsByDataProvider(BackendBulkDataProvide
 		return associationIds;
 	}
 
-	public void removeAssociation(Long id, String dataProviderName, String md5sum) {
+	public AlleleGeneAssociation deprecateOrDeleteAssociation(Long id, Boolean throwApiError, String loadDescription, Boolean deprecate) {
 		AlleleGeneAssociation association = alleleGeneAssociationDAO.find(id);
-		if (association != null) {
-			association.setSubject(null);
-			Long noteId = null;
-			if (association.getRelatedNote() != null)
-				noteId = association.getRelatedNote().getId();
-			alleleGeneAssociationDAO.remove(id);
-			if (noteId != null)
-				noteDAO.remove(noteId);
-		} else {
-			log.error("Failed getting allele-gene association: " + id);
+		
+		if (association == null) {
+			String errorMessage = "Could not find AlleleGeneAssociation with id: " + id;
+			if (throwApiError) {
+				ObjectResponse<AlleleGeneAssociation> response = new ObjectResponse<>();
+				response.addErrorMessage("id", errorMessage);
+				throw new ApiErrorException(response);
+			}
+			log.error(errorMessage);
+			return null;
+		}
+		if (deprecate) {
+			if (!association.getObsolete()) {
+				association.setObsolete(true);
+				if (authenticatedPerson.getOktaEmail() != null) {
+					association.setUpdatedBy(personService.findPersonByOktaEmail(authenticatedPerson.getOktaEmail()));
```

Comment:
You should be able to use the authenticatedPerson as the object for setUpdatedBy... if not you should be able to use the authenticatedPerson.id for a personDao.get(id)

Reply:
Updated here and in the DiseaseAnnotationService